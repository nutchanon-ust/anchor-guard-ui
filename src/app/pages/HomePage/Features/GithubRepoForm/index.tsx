import React, { useEffect } from 'react';
import styled from 'styled-components/macro';
import { useSelector, useDispatch } from 'react-redux';
import { FormLabel } from 'app/components/FormLabel';
import { Input } from './components/Input';
import { RepoItem } from './RepoItem';
import { TextButton } from './components/TextButton';
import {
  selectUsername,
  selectRepos,
  selectLoading,
  selectError,
} from './slice/selectors';
import { LoadingIndicator } from 'app/components/LoadingIndicator';
import { RepoErrorType } from './slice/types';
import { useGithubRepoFormSlice } from './slice';

export function GithubRepoForm() {
  const { actions } = useGithubRepoFormSlice();

  const username = useSelector(selectUsername);
  const repos = useSelector(selectRepos);
  const isLoading = useSelector(selectLoading);
  const error = useSelector(selectError);

  const dispatch = useDispatch();

  const onChangeUsername = (evt: React.ChangeEvent<HTMLInputElement>) => {
    dispatch(actions.changeUsername(evt.currentTarget.value));
    dispatch(actions.loadRepos());
  };

  const useEffectOnMount = (effect: React.EffectCallback) => {
    // eslint-disable-next-line react-hooks/exhaustive-deps
    useEffect(effect, []);
  };

  useEffectOnMount(() => {
    // When initial state username is not null, submit the form to load repos
    if (username && username.trim().length > 0) {
      dispatch(actions.loadRepos());
    }
  });

  const onSubmitForm = (evt?: React.FormEvent<HTMLFormElement>) => {
    /* istanbul ignore next  */
    if (evt !== undefined && evt.preventDefault) {
      evt.preventDefault();
    }
  };

  return (
    <Wrapper>
      <FormGroup onSubmit={onSubmitForm}>
        <FormLabel>Github Username</FormLabel>
        <InputWrapper>
          <Input
            type="text"
            placeholder="Type any Github username"
            value={username}
            onChange={onChangeUsername}
          />
          {isLoading && <LoadingIndicator small />}
        </InputWrapper>
      </FormGroup>
      {repos?.length > 0 ? (
        <List>
          {repos.map(repo => (
            <RepoItem
              key={repo.id}
              name={repo.name}
              starCount={repo.stargazers_count}
              url={repo.html_url}
            />
          ))}
        </List>
      ) : error ? (
        <ErrorText>{repoErrorText(error)}</ErrorText>
      ) : null}
    </Wrapper>
  );
}

export const repoErrorText = (error: RepoErrorType) => {
  switch (error) {
    case RepoErrorType.USER_NOT_FOUND:
      return 'There is no such user ðŸ˜ž';
    case RepoErrorType.USERNAME_EMPTY:
      return 'Type any Github username';
    case RepoErrorType.USER_HAS_NO_REPO:
      return 'User has no repository ðŸ¥º';
    case RepoErrorType.GITHUB_RATE_LIMIT:
      return 'Looks like github api`s rate limit(60 request/h) has exceeded ðŸ¤”';
    default:
      return 'An error has occurred!';
  }
};

const Wrapper = styled.div`
  ${TextButton} {
    margin: 16px 0;
    font-size: 0.875rem;
  }
`;

const InputWrapper = styled.div`
  display: flex;
  align-items: center;

  ${Input} {
    width: ${100 / 3}%;
    margin-right: 0.5rem;
  }
`;

const ErrorText = styled.span`
  color: ${p => p.theme.text};
`;

const FormGroup = styled.form`
  display: flex;
  flex-direction: column;
  margin-bottom: 1rem;

  ${FormLabel} {
    margin-bottom: 0.25rem;
    margin-left: 0.125rem;
  }
`;

const List = styled.div``;
