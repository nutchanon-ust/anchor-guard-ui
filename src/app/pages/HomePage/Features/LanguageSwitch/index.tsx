import * as React from 'react';
import { FormLabel } from 'app/components/FormLabel';
import { Radio } from 'app/components/Radio';
import styled from 'styled-components/macro';
import { useTranslation } from 'react-i18next';
import { messages } from './messages';

export function LanguageSwitch() {
  const { t, i18n } = useTranslation();
  const handleLanguageChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const language = event.target.value;
    i18n.changeLanguage(language);
  };

  return (
    <Wrapper>
      <FormLabel>{t(...messages.selectLanguage())}</FormLabel>
      <Languages>
        <Radio
          id="en"
          label="English"
          className="radio"
          name="language"
          onChange={handleLanguageChange}
          value="en"
          isSelected={i18n.language === 'en'}
        />
        <Radio
          id="tr"
          label="Deutsch"
          className="radio"
          name="language"
          onChange={handleLanguageChange}
          value="de"
          isSelected={i18n.language === 'de'}
        />
      </Languages>
    </Wrapper>
  );
}

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  ${FormLabel} {
    margin-bottom: 0.625rem;
  }
`;
const Languages = styled.div`
  display: flex;

  .radio {
    margin-right: 1.5rem;
  }
`;
