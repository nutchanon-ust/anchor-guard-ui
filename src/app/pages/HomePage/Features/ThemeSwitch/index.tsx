import * as React from 'react';
import { FormLabel } from 'app/components/FormLabel';
import { Radio } from 'app/components/Radio';
import styled from 'styled-components/macro';
import { themeActions } from 'styles/theme/slice';
import { useDispatch, useSelector } from 'react-redux';
import { saveTheme } from 'styles/theme/utils';
import { ThemeKeyType } from 'styles/theme/slice/types';
import { selectThemeKey } from 'styles/theme/slice/selectors';

export function ThemeSwitch() {
  const theme = useSelector(selectThemeKey);
  const dispatch = useDispatch();

  const handleThemeChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const value = event.target.value as ThemeKeyType;
    saveTheme(value);
    dispatch(themeActions.changeTheme(value));
  };

  return (
    <Wrapper>
      <FormLabel>Select Theme</FormLabel>
      <Themes>
        <Radio
          id="system"
          label="System theme"
          className="radio"
          name="theme"
          onChange={handleThemeChange}
          value="system"
          isSelected={theme === 'system'}
        />
        <Radio
          id="light"
          label="Light"
          className="radio"
          name="theme"
          onChange={handleThemeChange}
          value="light"
          isSelected={theme === 'light'}
        />
        <Radio
          id="dark"
          label="Dark"
          className="radio"
          name="theme"
          onChange={handleThemeChange}
          value="dark"
          isSelected={theme === 'dark'}
        />
      </Themes>
    </Wrapper>
  );
}

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  ${FormLabel} {
    margin-bottom: 0.625rem;
  }
`;
const Themes = styled.div`
  display: flex;

  .radio {
    margin-right: 1.5rem;
  }
`;
