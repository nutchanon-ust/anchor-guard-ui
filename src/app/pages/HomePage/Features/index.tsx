import * as React from 'react';
import styled from 'styled-components/macro';
import { Title } from '../components/Title';
import { Lead } from '../components/Lead';
import { SubTitle } from '../components/SubTitle';
import { P } from '../components/P';
import { A } from 'app/components/A';
import { GithubRepoForm } from './GithubRepoForm';
import { ThemeSwitch } from './ThemeSwitch';
import { LanguageSwitch } from './LanguageSwitch';
import { ReactComponent as StateIcon } from './assets/state.svg';
import { ReactComponent as CSSIcon } from './assets/css.svg';
import { ReactComponent as INTLIcon } from './assets/intl.svg';
import { ReactComponent as TSLogo } from './assets/ts.svg';
import { ReactComponent as RouteIcon } from './assets/route.svg';
import { ReactComponent as SEOIcon } from './assets/seo.svg';
import { ReactComponent as InstantFeedbackIcon } from './assets/instant-feedback.svg';
import { ReactComponent as ScaffoldingIcon } from './assets/scaffolding.svg';
import { ReactComponent as CodeAnalysisIcon } from './assets/code-analysis.svg';
import { useTranslation } from 'react-i18next';
import { Link } from 'app/components/Link';
import { messages } from '../messages';

export function Features() {
  const { t } = useTranslation();

  return (
    <>
      <Title as="h2">Features</Title>
      <Lead>
        Crafted for <strong>highly scalable</strong>,{' '}
        <strong>easily maintainable</strong> and{' '}
        <strong>highly performant</strong> React.js applications with a focus on{' '}
        <strong>best DX</strong> and <strong>best practices</strong>.
      </Lead>
      <List>
        <Feature>
          <StateIcon className="feature-icon" />
          <Content>
            <SubTitle>Predictable State</SubTitle>
            <P>
              Build easy to test and debug, flexible and extensible applications
              using{' '}
              <A
                href="https://redux.js.org/basics/usage-with-react"
                target="_blank"
                rel="noopener noreferrer"
              >
                Redux
              </A>
              . Unidirectional data flow allows for change logging and time
              travel debugging.{' '}
              <A
                href="https://github.com/zalmoxisus/redux-devtools-extension"
                target="_blank"
                rel="noopener noreferrer"
              >
                Install Chrome Redux Dev Tools
              </A>{' '}
              to see how your application's state changes and travel in time to
              debug. Type any github username below and see it in action with
              Redux Dev Tools.
            </P>
            <GithubRepoForm />
          </Content>
        </Feature>
        <Feature>
          <CSSIcon className="feature-icon" />
          <Content>
            <SubTitle>Next Generation CSS</SubTitle>
            <P>
              Write composable CSS thatâ€™s co-located with your components for
              complete modularity. Ship only the styles that are on the page for
              the best performance. Generate application-wide styles and themes
              for your components. Change the theme below to see how easy and
              intuitive theming can ever be!
            </P>
            <ThemeSwitch />
          </Content>
        </Feature>
        <Feature>
          <INTLIcon className="feature-icon" />
          <Content>
            <SubTitle>{t(messages.i18nTitle())} </SubTitle>
            <P>
              {t(messages.i18nDescription())}
              <br />
              <small>
                (Only some of the features below are translated to demonstrate
                an example)
              </small>
            </P>
            <LanguageSwitch />
          </Content>
        </Feature>
        <Feature>
          <RouteIcon className="feature-icon" />
          <Content>
            <SubTitle>{t(...messages.routingTitle())}</SubTitle>
            <P>
              {t(messages.routingDescription())}
              <br />
              <small>
                Go to our{' '}
                <Link to={process.env.PUBLIC_URL + '/notfound'}>/NotFound</Link>{' '}
                page to see how routing works
              </small>
            </P>
          </Content>
        </Feature>
        <Feature>
          <InstantFeedbackIcon className="feature-icon" />
          <Content>
            <SubTitle>{t(messages.feedbackTitle())}</SubTitle>
            <P>{t(messages.feedbackDescription())}</P>
          </Content>
        </Feature>
        <Feature>
          <ScaffoldingIcon className="feature-icon" />
          <Content>
            <SubTitle>{t(messages.scaffoldingTitle())}</SubTitle>
            <P>{t(messages.scaffoldingDescription())}</P>
          </Content>
        </Feature>
        <Feature>
          <SEOIcon className="feature-icon" />
          <Content>
            <SubTitle>SEO</SubTitle>
            <P>
              Supports SEO (document head tags management) for search engines
              that support indexing of JavaScript content.
            </P>
          </Content>
        </Feature>
        <Feature>
          <TSLogo className="feature-icon" />
          <Content>
            <SubTitle>TypeScript</SubTitle>
            <P>
              Typescript is the key to scalability. Build self-documented code,
              easy-to-debug code and create maintainable large applications and
              codebases with a highly productive development experience.
            </P>
          </Content>
        </Feature>
        <Feature>
          <CodeAnalysisIcon className="feature-icon" />
          <Content>
            <SubTitle>Static Code Analysis</SubTitle>
            <P>
              Focus on writing new features without worrying about formatting or
              code quality. With the right editor setup, your code will
              automatically be formatted and linted as you work.
            </P>
          </Content>
        </Feature>
      </List>
    </>
  );
}

const Feature = styled.li`
  display: flex;
  margin: 6.25rem 0 6.25rem 2.25rem;

  .feature-icon {
    width: 6.25rem;
    height: 6.25rem;
    margin-right: 2.25rem;
    flex-shrink: 0;
  }
`;
const Content = styled.div`
  flex: 1;
`;

const List = styled.ul`
  padding: 0;
  margin: 6.25rem 0 0 0;
`;
