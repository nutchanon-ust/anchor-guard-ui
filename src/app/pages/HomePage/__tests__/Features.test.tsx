import * as React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import { Features } from '../Features';
import i18next from 'i18next';

const shallowRenderer = createRenderer();

describe('<Features />', () => {
  it.only('should render with en translations', () => {
    i18next.changeLanguage('en');
    shallowRenderer.render(<Features />);
    const renderedOutput = shallowRenderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });

  it('should render with de translations', () => {
    i18next.changeLanguage('de');
    shallowRenderer.render(<Features />);
    const renderedOutput = shallowRenderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });

  it('should render different after language change', () => {
    i18next.changeLanguage('en');
    shallowRenderer.render(<Features />);
    const renderedOutput1 = shallowRenderer.getRenderOutput();

    i18next.changeLanguage('de');
    shallowRenderer.render(<Features />);
    const renderedOutput2 = shallowRenderer.getRenderOutput();

    expect(renderedOutput1).not.toEqual(renderedOutput2);
  });
});
