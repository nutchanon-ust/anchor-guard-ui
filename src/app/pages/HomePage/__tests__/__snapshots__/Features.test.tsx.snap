// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Features /> should render with de translations 1`] = `
<React.Fragment>
  <Title
    as="h2"
  >
    Features
  </Title>
  <Lead>
    Crafted for 
    <strong>
      highly scalable
    </strong>
    ,
     
    <strong>
      easily maintainable
    </strong>
     and
     
    <strong>
      highly performant
    </strong>
     React.js applications with a focus on
     
    <strong>
      best DX
    </strong>
     and 
    <strong>
      best practices
    </strong>
    .
  </Lead>
  <Features__List>
    <Features__Feature>
      <ForwardRef(SvgState)
        className="feature-icon"
      />
      <Features__Content>
        <SubTitle>
          Predictable State
        </SubTitle>
        <P>
          Build easy to test and debug, flexible and extensible applications using
           
          <A
            href="https://redux.js.org/basics/usage-with-react"
            rel="noopener noreferrer"
            target="_blank"
          >
            Redux
          </A>
          . Unidirectional data flow allows for change logging and time travel debugging.
           
          <A
            href="https://github.com/zalmoxisus/redux-devtools-extension"
            rel="noopener noreferrer"
            target="_blank"
          >
            Install Chrome Redux Dev Tools
          </A>
           
          to see how your application's state changes and travel in time to debug. Type any github username below and see it in action with Redux Dev Tools.
        </P>
        <GithubRepoForm />
      </Features__Content>
    </Features__Feature>
    <Features__Feature>
      <ForwardRef(SvgCss)
        className="feature-icon"
      />
      <Features__Content>
        <SubTitle>
          Next Generation CSS
        </SubTitle>
        <P>
          Write composable CSS that’s co-located with your components for complete modularity. Ship only the styles that are on the page for the best performance. Generate application-wide styles and themes for your components. Change the theme below to see how easy and intuitive theming can ever be!
        </P>
        <ThemeSwitch />
      </Features__Content>
    </Features__Feature>
    <Features__Feature>
      <ForwardRef(SvgIntl)
        className="feature-icon"
      />
      <Features__Content>
        <SubTitle>
          i18n Internationalisierung und Pluralisierung
           
        </SubTitle>
        <P>
          Das Internet ist global. Mehrsprachige- und Pluralisierungsunterstützung ist entscheidend für große Web-Anwendungen. Sie können die Sprache unten verändern, ohne die Seite aktualisieren zu müssen.
          <br />
          <small>
            (Only some of the features below are translated to demonstrate an example)
          </small>
        </P>
        <LanguageSwitch />
      </Features__Content>
    </Features__Feature>
    <Features__Feature>
      <ForwardRef(SvgRoute)
        className="feature-icon"
      />
      <Features__Content>
        <SubTitle>
          Standard Routing
        </SubTitle>
        <P>
          Routing macht es möglich Seiten (z.B. '/about') Ihrer Anwendung hinzuzufügen.
          <br />
          <small>
            Go to our
             
            <Link
              to="/notfound"
            >
              /NotFound
            </Link>
             
            page to see how routing works
          </small>
        </P>
      </Features__Content>
    </Features__Feature>
    <Features__Feature>
      <ForwardRef(SvgInstantFeedback)
        className="feature-icon"
      />
      <Features__Content>
        <SubTitle>
          Sofortiges Feedback
        </SubTitle>
        <P>
          Genießen Sie die beste Entwicklungserfahrung und programmieren Sie Ihre App so schnell wie noch nie! Ihre Änderungen an dem CSS und JavaScript sind sofort reflektiert, ohne die Seite aktualisieren zu müssen.
        </P>
      </Features__Content>
    </Features__Feature>
    <Features__Feature>
      <ForwardRef(SvgScaffolding)
        className="feature-icon"
      />
      <Features__Content>
        <SubTitle>
          Schnelles Scaffolding
        </SubTitle>
        <P>
          Automatisieren Sie die Kreation von Komponenten, Containern, Routen, Selektoren und Sagas – und ihre Tests – direkt von dem Terminal!
        </P>
      </Features__Content>
    </Features__Feature>
    <Features__Feature>
      <ForwardRef(SvgSeo)
        className="feature-icon"
      />
      <Features__Content>
        <SubTitle>
          SEO
        </SubTitle>
        <P>
          Supports SEO (document head tags management) for search engines that support indexing of JavaScript content.
        </P>
      </Features__Content>
    </Features__Feature>
    <Features__Feature>
      <ForwardRef(SvgTs)
        className="feature-icon"
      />
      <Features__Content>
        <SubTitle>
          TypeScript
        </SubTitle>
        <P>
          Typescript is the key to scalability. Build self-documented code, easy-to-debug code and create maintainable large applications and codebases with a highly productive development experience.
        </P>
      </Features__Content>
    </Features__Feature>
    <Features__Feature>
      <ForwardRef(SvgCodeAnalysis)
        className="feature-icon"
      />
      <Features__Content>
        <SubTitle>
          Static Code Analysis
        </SubTitle>
        <P>
          Focus on writing new features without worrying about formatting or code quality. With the right editor setup, your code will automatically be formatted and linted as you work.
        </P>
      </Features__Content>
    </Features__Feature>
  </Features__List>
</React.Fragment>
`;

exports[`<Features /> should render with en translations 1`] = `
<React.Fragment>
  <styled.h1
    as="h2"
  >
    Features
  </styled.h1>
  <styled.p>
    Crafted for 
    <strong>
      highly scalable
    </strong>
    ,
     
    <strong>
      easily maintainable
    </strong>
     and
     
    <strong>
      highly performant
    </strong>
     React.js applications with a focus on
     
    <strong>
      best DX
    </strong>
     and 
    <strong>
      best practices
    </strong>
    .
  </styled.p>
  <styled.ul>
    <styled.li>
      <ForwardRef(SvgState)
        className="feature-icon"
      />
      <styled.div>
        <styled.h3>
          Predictable State
        </styled.h3>
        <styled.p>
          Build easy to test and debug, flexible and extensible applications using
           
          <styled.a
            href="https://redux.js.org/basics/usage-with-react"
            rel="noopener noreferrer"
            target="_blank"
          >
            Redux
          </styled.a>
          . Unidirectional data flow allows for change logging and time travel debugging.
           
          <styled.a
            href="https://github.com/zalmoxisus/redux-devtools-extension"
            rel="noopener noreferrer"
            target="_blank"
          >
            Install Chrome Redux Dev Tools
          </styled.a>
           
          to see how your application's state changes and travel in time to debug. Type any github username below and see it in action with Redux Dev Tools.
        </styled.p>
        <GithubRepoForm />
      </styled.div>
    </styled.li>
    <styled.li>
      <ForwardRef(SvgCss)
        className="feature-icon"
      />
      <styled.div>
        <styled.h3>
          Next Generation CSS
        </styled.h3>
        <styled.p>
          Write composable CSS that’s co-located with your components for complete modularity. Ship only the styles that are on the page for the best performance. Generate application-wide styles and themes for your components. Change the theme below to see how easy and intuitive theming can ever be!
        </styled.p>
        <ThemeSwitch />
      </styled.div>
    </styled.li>
    <styled.li>
      <ForwardRef(SvgIntl)
        className="feature-icon"
      />
      <styled.div>
        <styled.h3>
          i18n Internationalization & Pluralization
           
        </styled.h3>
        <styled.p>
          Scalable apps need to support multiple languages, easily add and support multiple languages. Change the language below to see how instantly it updates the page without refreshing.
          <br />
          <small>
            (Only some of the features below are translated to demonstrate an example)
          </small>
        </styled.p>
        <LanguageSwitch />
      </styled.div>
    </styled.li>
    <styled.li>
      <ForwardRef(SvgRoute)
        className="feature-icon"
      />
      <styled.div>
        <styled.h3>
          Industry-standard Routing
        </styled.h3>
        <styled.p>
          It's natural to want to add pages (e.g. \`/about\`) to your application, and routing makes this possible.
          <br />
          <small>
            Go to our
             
            <Styled(Link)
              to="/notfound"
            >
              /NotFound
            </Styled(Link)>
             
            page to see how routing works
          </small>
        </styled.p>
      </styled.div>
    </styled.li>
    <styled.li>
      <ForwardRef(SvgInstantFeedback)
        className="feature-icon"
      />
      <styled.div>
        <styled.h3>
          Instant Feedback
        </styled.h3>
        <styled.p>
          Enjoy the best DX and code your app at the speed of thought! Your saved changes to the CSS and JS are reflected instantaneously without refreshing the page.
        </styled.p>
      </styled.div>
    </styled.li>
    <styled.li>
      <ForwardRef(SvgScaffolding)
        className="feature-icon"
      />
      <styled.div>
        <styled.h3>
          Quick Scaffolding
        </styled.h3>
        <styled.p>
          Automate the creation of components, features, routes, selectors and sagas - and their tests - right from the CLI! Avoid fighting the glue of your code and focus on your app!
        </styled.p>
      </styled.div>
    </styled.li>
    <styled.li>
      <ForwardRef(SvgSeo)
        className="feature-icon"
      />
      <styled.div>
        <styled.h3>
          SEO
        </styled.h3>
        <styled.p>
          Supports SEO (document head tags management) for search engines that support indexing of JavaScript content.
        </styled.p>
      </styled.div>
    </styled.li>
    <styled.li>
      <ForwardRef(SvgTs)
        className="feature-icon"
      />
      <styled.div>
        <styled.h3>
          TypeScript
        </styled.h3>
        <styled.p>
          Typescript is the key to scalability. Build self-documented code, easy-to-debug code and create maintainable large applications and codebases with a highly productive development experience.
        </styled.p>
      </styled.div>
    </styled.li>
    <styled.li>
      <ForwardRef(SvgCodeAnalysis)
        className="feature-icon"
      />
      <styled.div>
        <styled.h3>
          Static Code Analysis
        </styled.h3>
        <styled.p>
          Focus on writing new features without worrying about formatting or code quality. With the right editor setup, your code will automatically be formatted and linted as you work.
        </styled.p>
      </styled.div>
    </styled.li>
  </styled.ul>
</React.Fragment>
`;
